lib/retry.ts

```diff
@@ -91,6 +91,20 @@ function validateRetryConfig<T>(config: RetryConfig<T>): void {
   }
 }
 
+
+async function checkInternet(): Promise<boolean> {
```

Comment:
I would rename this as `checkNetworkConnection` or `checkConnectionToAzure`

Reply:
+1 for `checkNetworkConnection` or `checkNetwork`